examples/stubbing-spying__route2/cypress/integration/stub-fetch-spec.js

```diff
@@ -0,0 +1,133 @@
+/// <reference types="Cypress" />
+
+describe('route2', () => {
+  context('stubbing', function () {
+    it('shows no Response message', () => {
+      // stub server response with []
+      cy.route2('/favorite-fruits', JSON.stringify([]))
+      cy.visit('/')
+      cy.contains('No favorites').should('be.visible')
+    })
+
+    it('stubs fetch to test loading indicator', () => {
+      cy.route2('/favorite-fruits', (req) => {
+        req.reply((res) => {
+          res.delay(2000).send(['Pineapple 🍍'])
+        })
+      })
+
+      cy.visit('/')
+      // at first, the app is showing the loading indicator
+      cy.get('.loader').should('be.visible')
+      // once the promise is resolved, the loading indicator goes away
+      cy.get('.loader').should('not.exist')
+      cy.contains('li', 'Pineapple 🍍')
+    })
+
+    // A big advantage of controlling the response is we can test
+    // how our app handles a slow response, which normally might be
+    // difficult against a fast development server
+    it('shows loader while fetching fruits', function () {
+    // stub the XHR request from the app
+      cy.route2('/favorite-fruits', (req) => {
+        req.reply((res) => {
+          // hmm, every time we want to return an empty list
+          // we need to stringify it, otherwise the stub does not ... stub
+          res.delay(1000).send(JSON.stringify([]))
+        })
+      })
+
+      cy.visit('/')
+      cy.get('.loader').should('be.visible')
+
+      // once the network call finishes, the loader goes away
+      cy.get('.loader').should('not.exist')
+      cy.contains('.favorite-fruits', 'No favorites')
+    })
+
+    it('can spy on network calls from the second page', () => {
+      cy.route2('/favorite-fruits').as('favoriteFruits')
+      cy.visit('/')
+      cy.wait('@favoriteFruits')
+
+      cy.contains('a', 'Go to page 2').click()
+      cy.url().should('match', /\/page2\.html$/)
+      // the second page also requests the fruits
+      cy.wait('@favoriteFruits')
+    })
+
+    it('can stub network calls for each page', () => {
+      let k = 0
+
+      cy.route2('/favorite-fruits', (req) => {
+        k += 1
+        switch (k) {
+          case 1:
+            return req.reply(['apples 🍎'])
+          case 2:
+            return req.reply(['grapes 🍇'])
+          default:
+            return req.reply(['kiwi 🥝'])
+        }
+      })
```

Comment:
i'm thinking `RouteMatcher` should have an additional field, something like `times: number`, where it specifies the number of times it will apply. then this would simply be:

```js
cy.route2({ url: '/favorite-fruits', times: 1 }, ['apples 🍎'])
cy.route2({ url: '/favorite-fruits', times: 1 }, ['grapes 🍇'])
cy.route2({ url: '/favorite-fruits', times: 1 }, ['kiwi 🥝'])
```

what do you think?

Reply:
opened an issue here, if you think of a better API feel free to comment: https://github.com/cypress-io/cypress/issues/8531

Reply:
what happens after that? does it continue stubbing with the last one? Does it stop stubbing and just does spying?

Reply:
Returns to the default proxy behavior - passthru